docs/elasticsearch/cluster.md

```diff
@@ -198,6 +189,135 @@ x.x.x.x           23          38   0    0.12    0.07     0.06 md        -      o
 
 To better understand and monitor your cluster, use the [cat API](../catapis/).
 
+## Step 6: Configure shard allocation awareness or forced awareness
+
+If your nodes are spread across many geographical zones, you can configure shard allocation awareness to allocate all replica shards to a zone that’s different from their primary shard.
+
+With shard allocation awareness, if the nodes in one of your zones fail, you can be assured that your replica shards are spread across your other zones. It adds a layer of fault tolerance than just individual node failure.
+
+To configure shard allocation awareness, add zone attributes to `odfe-d1` and `odfe-d2`, respectively:
+
+```yml
+node.attr.zone: 1
+```
+```yml
+node.attr.zone: 2
+```
+
+Log in to your master node and update the cluster settings:
+
+```json
+PUT _cluster/settings
+{
+  "persistent": {
+    "cluster.routing.allocation.awareness.attributes": "zone"
+  }
+}
+```
+
+You can either use `persistent` or `transient` settings. Persistent settings persist through a cluster reboot, transient settings do not.
+
+Shard allocation awareness prefers to separate primary and replica shards across multiple zones but if there is only one zone available like in the event of a failure, then replica shards are allocated to the only remaining zone.
+
+You also have the option to require that primary and replica shards are never allocated to the same zone. This is called forced awareness.
+
+To configure forced awareness, specify all the possible values for your zone attributes:
+
+```json
+PUT _cluster/settings
+{
+  "persistent": {
+    "cluster.routing.allocation.awareness.attributes": "zone",
+    "cluster.routing.allocation.awareness.force.zone.values":["1", "2"]
+  }
+}
+```
+Now, if one of the data nodes fail, forced awareness will not allocate the replicas to a node in the same zone. Instead, it will keep the cluster in a yellow state and only allocate replicas when nodes in another zone come back online.
+
+You generally want to use allocation awareness when the hardware in each of your zones is less than 50% utilized so that it has the storage capacity to allocate replicas in the same zone.
```

Comment:
Is this generalizable? What if I have five zones? It seems unlikely that I'd have four zones go down and that one zone would have to host its own replicas. So at two zones, sure, 50%, but at three, 66%, four, 75%, and five, 80%?

Then again, you also have to account for disk watermarks (default low watermark is 85%), and Linux uses 5%, so now we're talking about 50% of the remaining 80%, etc.

Basically, I suspect we need to go deeper into the weeds before this is good advice.

Reply:
Yes, it' not generalizable. Maybe we can just stick to our specific architecture and say that In our specific case, we can use allocation awareness if odfe-d1 and odfe-d2 are less than 50% utilized, so that each of them have the storage capacity to allocate replicas in the same zone.
If that is not the case, we can use forced awareness.

Using allocation awareness or forced awareness would depend on how you to want to distribute your shards across the nodes in your cluster.

Reply:
Cool, yeah, noting that it's specific to our two zone architecture here would be useful. Maybe some blanket statement about how you might need more or less space depending on how many zones you have and how you've distributed your nodes.

Reply:
How about "Using allocation awareness or forced awareness would depend on how much you space you might need in each zone to balance your primary and replica shards for cluster stability."?